src/persistence/EntityPersistExecutor.ts

```diff
@@ -73,14 +73,24 @@ export class EntityPersistExecutor {
                 // create subjects for all entities we received for the persistence
                 entities.forEach(entity => {
                     const entityTarget = this.target ? this.target : entity.constructor;
+                    const metadata = this.connection.getMetadata(entityTarget);
                     if (entityTarget === Object)
                         throw new CannotDetermineEntityError(this.mode);
 
+                    if (this.mode === "restore" && metadata.softDeletedDateColumn) {
+                        entity[metadata.softDeletedDateColumn.propertyName] = null;
```

Comment:
Please use `metadata.softDeletedDateColumn.setEntityValue(null)` instead. This is important if your column is in embedded.

Reply:
will be fixed in next update